services/api/src/hrrr_smoke/routes.py

```diff
@@ -43,67 +41,90 @@ def distance(start, end):
-        datetime.strptime(forecast, forecast_format) for forecast, _ in z.groups()
+        (datetime.strptime(run_hour, forecast_format), list(map(int, forecast.valid_time)))
+        for run_hour, forecast in z.groups()
     ]
 
-    forecast_list.sort()
+    current_app.logger.debug(
+        f"Loaded {len(forecast_list)} runs from {current_app.config.forecasts_array}"
+    )
 
-    return jsonify(
-        [
+    forecast_list.sort(key=lambda t: t[0])
+
+    response = [
             {
-                "iso": dt.isoformat(),
-                "forecast": dt.strftime(forecast_format),
-                "display": dt.strftime("%d %b %Y %H:%M:%S"),
+                "run_hour": run_hour.isoformat(),
+                "valid_times": valid_times,
             }
-            for dt in forecast_list
+            for run_hour, valid_times in forecast_list
         ]
-    )
+
+    try:
+        return jsonify(response)
+    except Exception as e:
+        current_app.logger.exception("Unable to serialize response")
+        raise e
+
 
 
 @bp.route("/xsection/")
 def xsection():
-    forecast = request.args["forecast"]
+    current_app.logger.debug("GET /xsection/")
+
+    run_hour = datetime.strptime(request.args["runHour"], "%Y-%m-%dT%H:%M:%S")
+    valid_time = int(request.args["validTime"])
+
+    current_app.logger.debug(f"runHour: {run_hour}")
+    current_app.logger.debug(f"validTime: {valid_time}")
 
     # Start and end points of the path for the cross-section (latitude,
     # longitude)
     start = (float(request.args["startLat"]), float(request.args["startLng"]))
     end = (float(request.args["endLat"]), float(request.args["endLng"]))
 
+    current_app.logger.debug(f"start: {start}")
+    current_app.logger.debug(f"end: {end}")
+
     # Number of steps taken along the path
     steps = 1200
 
-    dataset = xr.open_zarr(current_app.config.forecasts_array, group=forecast)
-    dataset = dataset.metpy.assign_crs(CF_ATTRS).metpy.parse_cf().squeeze()
+    dataset = xr.open_zarr(
+        f"{current_app.config.forecasts_array}/{run_hour.strftime('%Y%j%H%M%S')}"
+    )
+    dataset = dataset.isel(valid_time=valid_time).metpy.assign_crs(CF_ATTRS).metpy.parse_cf().squeeze()
+
+    current_app.logger.debug(f"dataset: {dataset}")
 
     cross = cross_section(dataset, start, end, steps).set_coords(
         ("latitude", "longitude")
     )
     plevs = np.arange(1000.0, 100, -20.0, dtype=np.float32) * units.hPa
-    temperature = units.Quantity(cross["t"].values, "degK")
-    pressure = units.Quantity(cross["pres"].values, "Pa")
-    potential_temperature = metpy.calc.potential_temperature(pressure, temperature)
-
-    potential_temperature, massden, temperature = log_interpolate_1d(
-        plevs,
-        pressure,
-        potential_temperature,
-        cross["massden"].values,
-        temperature,
-        axis=0,
-    )
+    pressure = units.Quantity(cross["isobaric"].values, "Pa")
+
+    try:
+        massden = log_interpolate_1d(
```

Comment:
@jqs-noaa I’m trying to update this endpoint to return the cross-section data from your Zarr arrays, but I’m getting an exception when I try to call `log_interpolate_1d`. There are two arrays with mismatched shapes, but I don’t really know what to do about it.

```
[2022-02-09 23:07:31,156] ERROR in routes: log_interpolate_1d failed
Traceback (most recent call last):
   File "/usr/src/app/src/hrrr_smoke/routes.py", line 114, in xsection
     axis=0,
   File "/usr/local/lib/python3.7/site-packages/metpy/xarray.py", line 1216, in wrapper
     result = func(*bound_args.args, **bound_args.kwargs)
   File "/usr/local/lib/python3.7/site-packages/metpy/interpolate/one_dimension.py", line 226, in log_interpolate_1d
     return interpolate_1d(log_x, log_xp, *args, axis=axis, fill_value=fill_value)
   File "/usr/local/lib/python3.7/site-packages/metpy/xarray.py", line 1216, in wrapper
     result = func(*bound_args.args, **bound_args.kwargs)
   File "/usr/local/lib/python3.7/site-packages/metpy/interpolate/one_dimension.py", line 158, in interpolate_1d
     / (xp[above] - xp[below]))
ValueError: operands could not be broadcast together with shapes (45,1200) (45,)
````